js/src/structured-data-blocks/how-to/components/HowTo.js

```diff
@@ -55,25 +55,31 @@ export default class HowTo extends Component {
 	/**
 	 * Replaces the How-to step with the given index.
 	 *
-	 * @param {array|string} newName      The new contents of the step-name.
-	 * @param {array|string} newText      The new contents of the step-text.
-	 * @param {array|string} previousName The previous contents of the step-name.
-	 * @param {array|string} previousText The previous contents of the step-text.
-	 * @param {number}       index        The index of the step that needs to be changed.
+	 * @param {array} newName      The new step-name.
+	 * @param {array} newText      The new step-text.
+	 * @param {array} previousName The previous step-name.
+	 * @param {array} previousText The previous step-text.
+	 * @param {number} index       The index of the step that needs to be changed.
 	 *
 	 * @returns {void}
 	 */
 	changeStep( newName, newText, previousName, previousText, index ) {
 		let steps = this.props.attributes.steps ? this.props.attributes.steps.slice() : [];
 
+		// If the index exceeds the amount of steps, don't change anything.
 		if ( index >= steps.length ) {
 			return;
 		}
 
+		/*
+		 * If the name of the step and the previous name don't match or if the text of the step and the text of
+		 * the previous step don't match, don't change anything.
+		 */
```

Comment:
I enjoy the spirit of this CR implementation ("please add comment what's happening here"), but I figured @IreneStr meant a more mechanical explanation on the "why" rather than a code to sentence conversion. ^^

It's a bit funky, the why. I'll provide it on Monday if still needed. But I agree with @afercia that these things could be abstracted anyways, so probably comment it in the new functions.